spectrum-network/src/peer_manager.rs

```diff
@@ -32,11 +32,23 @@ pub enum PeerManagerOut {
     /// Drop the connection to the given peer, or cancel the connection attempt after a `Connect`.
     Drop(PeerId),
     /// Approves an incoming connection.
-    Accept(PeerId, ConnectionId),
+    AcceptIncomingConnection(PeerId, ConnectionId),
+    /// Establish an outgoing connection.
+    EstablishOutgoingConnection(PeerId, ConnectionId),
```

Comment:
What's the purpose of this command? How is this different from `Connect`?

Reply:
`EstablishOutgoingConnection` is sent when a `ConnectedPeer` calls `confirm_connection()` (this happens in `PeerManager::on_connection_established`). For your convenience this is how `confirm_connection` is defined:
<img width="697" alt="image" src="https://user-images.githubusercontent.com/791016/203216412-ae7fcc1c-1f9d-44aa-9443-5d85b0ea007d.png">


Now `PeerManagerOut::Connect` is issued in `PeerManager::connect()`, which leads to NC performing a `NetworkBehaviourAction::Dial` action.

Reply:
Please disregard the previous comment. `EstablishOutgoingConnection` variant is not needed at all.